homeassistant/components/notify/slack.py

```diff
@@ -22,6 +23,8 @@
 PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
     vol.Required(CONF_API_KEY): cv.string,
     vol.Required(CONF_CHANNEL): cv.string,
+    vol.Optional(CONF_USERNAME, default=None): cv.string,
+    vol.Optional(CONF_ICON, default=None): cv.string,
```

Comment:
This will error if default value is used, when passed to `cv. string`.  Remove the default value,  and use dictionary lookup below.